### ISSUE TITLE
[defect]: TLS 1.0 support broken in FreeRADIUS 4.0.x

### ISSUE THREAD

**Issue by flux-inductor** [FreeRADIUS/freeradius-server#4191]
### What type of defect/bug is this?

Unexpected behaviour (obvious or verified by project member)

### How can the issue be reproduced?

Setting
                tls_min_version = "1.0"
                tls_max_version = "1.0"
in mods-enabled/eap resulted in a broken EAP-TLS configuration that refused to authenticate legacy clients requiring TLS 1.0. The parsing of tls_min_version = 1.0 is broken in ctx.c.

The failure would display references to TLS 1.3 in debugging output which made little sense:

(1)    authenticate eap {
(1)      eap - Continuing EAP session
(1)      eap - Peer sent packet with EAP method TLS (13)
(1)      eap - Calling submodule eap_tls
(1)      subrequest {
(1.0)      eap.tls - Continuing EAP-TLS
(1.0)      eap.tls - Peer indicated complete TLS record size will be 117 bytes
(1.0)      eap.tls - Got complete TLS record, with length field (117 bytes)
(1.0)      eap.tls - [eap-tls verify] = complete
(1.0)      Handshake state - before SSL initialization (0)
(1.0)      >>> send TLS 1.3, alert[length 2], fatal, internal_error
(1.0)      ERROR: Sending client fatal TLS alert: internal error 80
(1.0)      ERROR: Handshake exit state Server error
(1.0)      ERROR: Failed in SSL_read: error:141FC044:SSL routines:tls_setup_handshake:internal error
(1.0)      ERROR: System call (I/O) error (-1)
(1.0)      ERROR: TLS receive handshake failed during operation
(1.0)      eap.tls - ERROR: [eap-tls process] = fail
(1.0)      eap.tls (reject)
(1)        subrequest - Resuming execution
(1)      } # subrequest (noop)
(1)      eap - Sending EAP Failure (code 4) ID 89 length 4
(1)      eap - Cleaning up EAP session
(1)      eap (reject)
(1)    } # authenticate eap (reject)
(1)    radius - Failed to authenticate the user

I noticed in the debugging output a complaint message was displayed on startup for tls_max_version but not tls_min_version which was a clue something was wrong:

tls - TLS 1.0 and 1.1 are insecure and SHOULD NOT be used
tls - tls_max_version SHOULD be 1.2 or greater

There appears to have been a breaking change merged in commit 0568698c7656a048d1228d21f2070e06d135d19c specifically lines 616 & 630 in ctx.c which sets the default to TLS 1.2 but removes the else branch of the if statement:
+		int min_version = TLS1_2_VERSION;
-		else min_version = TLS1_VERSION;

Basically when both tls_max_version = 1.0 & tls_min_version = 1.0 in the eap config, you have a situation where                 tls_max_version = 1.0 but tls_min_version defaults to 1.2 in the code resulting in unpredictable behavior because tls_min_version = 1.0 has no matching conditional.

Restoring the line `else min_version = TLS1_VERSION` in ctx.c & recompiling fixed the issue and the client could connect.

### Log output from the FreeRADIUS daemon

```shell
# /usr/local/sbin/radiusd -v
Info  : radiusd - DEVELOPER BUILD - FreeRADIUS version 4.0.0-25523 (git #d4464d03), for host x86_64-unknown-linux-gnu, built on Aug 17 2021 at 17:59:47
Info  : FreeRADIUS Version 4.0.0
```

**Comment by arr2036** [comment]
I hit this with a customer as well.  We need to support TLS 1.0 for the time being.
